python_modules/dagster/dagster/_core/definitions/utils.py

```diff
@@ -311,3 +320,88 @@ def resolve_automation_condition(
 def dedupe_object_refs(objects: Optional[Iterable[T]]) -> Sequence[T]:
     """Dedupe definitions by reference equality."""
     return list({id(obj): obj for obj in objects}.values()) if objects is not None else []
+
+
+def add_default_automation_condition_sensor(
+    sensors: Sequence["SensorDefinition"],
+    assets: Iterable[Union["AssetsDefinition", "SourceAsset"]],
+    asset_checks: Iterable["AssetsDefinition"],
+) -> Sequence["SensorDefinition"]:
+    """Adds a default automation condition sensor if the provided sensors do not already handle all
+    provided assets.
+    """
+    from dagster._core.definitions.asset_graph import AssetGraph
+    from dagster._core.definitions.automation_condition_sensor_definition import (
+        DEFAULT_AUTOMATION_CONDITION_SENSOR_NAME,
+        AutomationConditionSensorDefinition,
+    )
+
+    asset_graph = AssetGraph.from_assets([*assets, *asset_checks])
+    sensor_selection = get_default_automation_condition_sensor_selection(sensors, asset_graph)
+    if sensor_selection:
+        default_sensor = AutomationConditionSensorDefinition(
+            DEFAULT_AUTOMATION_CONDITION_SENSOR_NAME, asset_selection=sensor_selection
+        )
+        sensors = [*sensors, default_sensor]
+
+    return sensors
+
+
+def get_default_automation_condition_sensor_selection(
```

Comment:
This ports over the logic from the old code in external.py, but I took the opportunity to simplify it a bit.

Reply:
Would recommend doing simplification in a follow up PR to reduce risk